rflx/model/message.py

```diff
@@ -1632,7 +1679,28 @@ def prune_dangling_fields(
                 break
 
             field, inner_message = inner_messages.pop(0)
-            inner_message = inner_message.prefixed(f"{field.name}_")
+            inner_message, message_attribute_locations = replace_message_attributes(
+                inner_message.prefixed(f"{field.name}_")
+            )
+
+            message_next_fields = message.outgoing(field)
+            if len(message_attribute_locations) > 0 and (
+                len(message_next_fields) > 1 or message_next_fields[0].target != FINAL
+            ):
+                self.error.append(
+                    "Message types with message attribute may only be used for fields preceeding"
+                    " FINAL",
+                    Subsystem.MODEL,
+                    Severity.ERROR,
+                    field.identifier.location,
+                )
+                for loc in message_attribute_locations:
+                    self.error.append(
+                        f"Message attribute used in {inner_message.identifier}",
```

Comment:
Error messages do not follow [design guide](https://github.com/Componolit/RecordFlux/blob/main/doc/Design-Guide.md).

Reply:
It's hard to always follow every single rule in the design guide ;-) Changed in 877fd78d.